test/fuzzyDateTime.js

```diff
@@ -0,0 +1,205 @@
+import {
+	formatFuzzyDate,
+	formatFuzzyDateTime,
+	millisecondsPer
+} from '../lib/fuzzyDateTime.js';
+
+const { second, minute, hour, day } = millisecondsPer;
+
+var expect = chai.expect;
+
+const noon = new Date(2000, 1, 1, 12, 0, 0, 0);
+const ninePM = new Date(2000, 1, 1, 21, 0, 0, 0);
+const threeAM = new Date(2000, 1, 1, 3, 0, 0, 0);
+const offset = (ms, origin = noon) => new Date(origin.getTime() + ms);
+const label = makeTime => `${makeTime}`.slice(6);
+
+const future = {
+	seconds: n => offset(n * second),
+	minutes: n => offset(n * minute),
+	hours: (n, origin = noon) => offset(n * hour, origin),
+	days: n => offset(n * day),
+};
+
+const past = {
+	seconds: (n, origin = noon) => offset(-n * second, origin),
+	minutes: (n, origin = noon) => offset(-n * minute, origin),
+	hours: (n, origin = noon) => offset(-n * hour, origin),
+	days: (n, origin = noon) => offset(-n * day, origin),
+};
+
+const shortDateRegex = /^\d{1,2}\/\d{1,2}\/\d{4}$/;
+const shortDateTimeRegex = /^\d{1,2}\/\d{1,2}\/\d{4} \d{1,2}:\d{2} [AP]M$/;
+const fullDateRegex = /^\w+, \w+ \d{1,2}, \d{4}$/;
+const fullDateTimeRegex = /^\w+, \w+ \d{1,2}, \d{4} \d{1,2}:\d{2} [AP]M\s*$/;
+
+const enAlwaysShort = new Intl.RelativeTimeFormat('en', {
+	localeMatcher: 'best fit',
+	numeric: 'always',
+	style: 'short'
+});
+
+const frAutoLong = new Intl.RelativeTimeFormat('fr', {
+	localeMatcher: 'best fit',
+	numeric: 'auto',
+	style: 'long'
+});
+
+const relativeExamples = [
+	[() => past.seconds(1), '1 second ago', () => 29 * second],
+	[() => past.seconds(29), '29 seconds ago', () => 1 * second],
+	[() => future.seconds(1), 'in 1 second', () => 31 * second],
+	[() => future.seconds(28), 'in 28 seconds', () => 58 * second],
```

Comment:
More new behaviour of onUpdate timeouts when time difference is under 30 seconds